### PR TITLE
Custom domain: add TLS cert requirements

### PR DESCRIPTION
**Pull request by brentschaus-okta**
## Description:
- **What's changed?** Adds a list of TLS cert requirements.
- **Is this PR related to a Monolith release?** Nope.

### Resolves:

* [OKTA-351249](https://oktainc.atlassian.net/browse/OKTA-351249)

### CODE REVIEW

**Comment by acrolinxdevdocs-okta** [comment]
## Acrolinx Scorecards

Acrolinx checked the content changed here.

Please review the scorecards for each file and make the recommended
improvements.

| Article | Score | Issues | Scorecard | Status |
| ------- | ----- | ------ | --------- | ------ |
| [packages/@okta/vuepress-site/docs/guides/custom-url-domain/main/index.md](https://github.com/okta/okta-developer-docs/blob/8e225457a8d83f8b1e097416f0634690b0777b4d/packages%2F%40okta%2Fvuepress-site%2Fdocs%2Fguides%2Fcustom-url-domain%2Fmain%2Findex.md) | 80 | 123 | [→](https://okta.acrolinx.cloud/api/v1/checking/scorecards/42645782-59d4-41e6-9060-59d8860986e5) | [:white_check_mark:](# "Successfully processed") |

Successfully checked 1 of 1 documents.
See summary in [Content Analysis Dashboard](https://okta.acrolinx.cloud/api/batch/gen.github.e87316b8-1d14-450f-9dad-8adac4b6b089?csBaseUrl=https%3A%2F%2Fokta.acrolinx.cloud&X-Acrolinx-Client-Locale=en)

Reopen the pull request or push new changes to check again.

Depending on the Acrolinx server configuration, the
links expire after some time and you must have a login for the
Acrolinx server to access them again.

**Comment by acrolinxdevdocs-okta** [comment]
## Acrolinx Scorecards

Acrolinx checked the content changed here.

Please review the scorecards for each file and make the recommended
improvements.

| Article | Score | Issues | Scorecard | Status |
| ------- | ----- | ------ | --------- | ------ |
| [packages/@okta/vuepress-site/docs/guides/custom-url-domain/main/index.md](https://github.com/okta/okta-developer-docs/blob/05b1cd6e0a9de22c4bb01136629111356383a63f/packages%2F%40okta%2Fvuepress-site%2Fdocs%2Fguides%2Fcustom-url-domain%2Fmain%2Findex.md) | 80 | 124 | [→](https://okta.acrolinx.cloud/api/v1/checking/scorecards/dd4ebfe1-b9bb-4c01-bff8-d4c858ae2b8a) | [:white_check_mark:](# "Successfully processed") |

Successfully checked 1 of 1 documents.
See summary in [Content Analysis Dashboard](https://okta.acrolinx.cloud/api/batch/gen.github.d945e158-a2c0-4848-8947-db99cbad713c?csBaseUrl=https%3A%2F%2Fokta.acrolinx.cloud&X-Acrolinx-Client-Locale=en)

Reopen the pull request or push new changes to check again.

Depending on the Acrolinx server configuration, the
links expire after some time and you must have a login for the
Acrolinx server to access them again.

File: packages/@okta/vuepress-site/docs/guides/custom-url-domain/main/index.md [hunk 1]
```diff
@@ -64,9 +64,27 @@ The third generation of the Okta Sign-In Widget doesn’t guarantee the stabilit
...
-    * If you have an Okta-managed certificate and you later get a CAA record, Okta can't renew your certificate. You must either add `letsencrypt.org` to the issuers list or remove the CAA record.
+  * If you have an Okta-managed certificate and you later get a CAA record, Okta can't renew your certificate. You must either add `letsencrypt.org` to the issuers list or remove the CAA record.
+
+* If you use your own TLS certificate, keep in mind the following:
+
+  * It should be signed with the SHA-256 hash algorithm.
+
+  * It shouldn't be expired.
+
+  * It's start date shouldn't be in the future.
```
Comment (gormanho-okta) [inline packages/@okta/vuepress-site/docs/guides/custom-url-domain/main/index.md:77]:
```suggestion
  * Its start date shouldn't be in the future.
```

File: packages/@okta/vuepress-site/docs/guides/custom-url-domain/main/index.md [hunk 2]
```diff
@@ -64,9 +64,27 @@ The third generation of the Okta Sign-In Widget doesn’t guarantee the stabilit
...
+
+  * It's expiration date can't be further than 10 years from now.
```
Comment (gormanho-okta) [inline packages/@okta/vuepress-site/docs/guides/custom-url-domain/main/index.md:79]:
```suggestion
  * Its expiration date can't be further than 10 years from now.
```

**Review by gormanho-okta — APPROVED** [review]

**Comment by acrolinxdevdocs-okta** [comment]
## Acrolinx Scorecards

Acrolinx checked the content changed here.

Please review the scorecards for each file and make the recommended
improvements.

| Article | Score | Issues | Scorecard | Status |
| ------- | ----- | ------ | --------- | ------ |
| [packages/@okta/vuepress-site/docs/guides/custom-url-domain/main/index.md](https://github.com/okta/okta-developer-docs/blob/58843f96fc1d194f38f5f73e67897aa9d2ec8f57/packages%2F%40okta%2Fvuepress-site%2Fdocs%2Fguides%2Fcustom-url-domain%2Fmain%2Findex.md) | 80 | 123 | [→](https://okta.acrolinx.cloud/api/v1/checking/scorecards/2fe87f8c-5ab3-44c4-993c-03de691f4c7e) | [:white_check_mark:](# "Successfully processed") |

Successfully checked 1 of 1 documents.
See summary in [Content Analysis Dashboard](https://okta.acrolinx.cloud/api/batch/gen.github.22674595-21f6-439e-9bdf-b2b49b410909?csBaseUrl=https%3A%2F%2Fokta.acrolinx.cloud&X-Acrolinx-Client-Locale=en)

Reopen the pull request or push new changes to check again.

Depending on the Acrolinx server configuration, the
links expire after some time and you must have a login for the
Acrolinx server to access them again.

**Comment by acrolinxdevdocs-okta** [comment]
## Acrolinx Scorecards

Acrolinx checked the content changed here.

Please review the scorecards for each file and make the recommended
improvements.

| Article | Score | Issues | Scorecard | Status |
| ------- | ----- | ------ | --------- | ------ |
| [packages/@okta/vuepress-site/docs/guides/custom-url-domain/main/index.md](https://github.com/okta/okta-developer-docs/blob/7fb9ed63ae80d04ac11270c4908dcbbc3333f111/packages%2F%40okta%2Fvuepress-site%2Fdocs%2Fguides%2Fcustom-url-domain%2Fmain%2Findex.md) | 80 | 122 | [→](https://okta.acrolinx.cloud/api/v1/checking/scorecards/f3350dac-0b44-4b07-8258-14661f9ae176) | [:white_check_mark:](# "Successfully processed") |

Successfully checked 1 of 1 documents.
See summary in [Content Analysis Dashboard](https://okta.acrolinx.cloud/api/batch/gen.github.3b3297e0-2ba2-46a6-9e2b-88ff49d7b320?csBaseUrl=https%3A%2F%2Fokta.acrolinx.cloud&X-Acrolinx-Client-Locale=en)

Reopen the pull request or push new changes to check again.

Depending on the Acrolinx server configuration, the
links expire after some time and you must have a login for the
Acrolinx server to access them again.

File: packages/@okta/vuepress-site/docs/guides/custom-url-domain/main/index.md [hunk 3]
```diff
@@ -64,9 +64,27 @@ The third generation of the Okta Sign-In Widget doesn’t guarantee the stabilit
...
 
-    * If it's your first time setting up a custom domain with an Okta-managed certificate, you need to add `letsencrypt.org` to the issuers list or Okta can't get the TLS certificate. See [Let's Encrypt - Using CAA](https://letsencrypt.org/docs/caa/).
+  * If it's your first time setting up a custom domain with an Okta-managed certificate, you need to add `letsencrypt.org` to the issuers list or Okta can't get the TLS certificate. See [Let's Encrypt - Using CAA](https://letsencrypt.org/docs/caa/).
 
-    * If you have an Okta-managed certificate and you later get a CAA record, Okta can't renew your certificate. You must either add `letsencrypt.org` to the issuers list or remove the CAA record.
+  * If you have an Okta-managed certificate and you later get a CAA record, Okta can't renew your certificate. You must either add `letsencrypt.org` to the issuers list or remove the CAA record.
+
+* If you use your own TLS certificate, keep in mind the following:
```
Comment (timhowie-okta) [inline packages/@okta/vuepress-site/docs/guides/custom-url-domain/main/index.md:71]: would 'consider' be easier for non-english speakers to understand than 'keep in mind', which might be a phrase that might be more difficult to parse?

File: packages/@okta/vuepress-site/docs/guides/custom-url-domain/main/index.md [hunk 4]
```diff
@@ -64,9 +64,27 @@ The third generation of the Okta Sign-In Widget doesn’t guarantee the stabilit
...
+
+  * It should be signed with the SHA-256 hash algorithm.
```
Comment (timhowie-okta) [inline packages/@okta/vuepress-site/docs/guides/custom-url-domain/main/index.md:73]: should or must?

File: packages/@okta/vuepress-site/docs/guides/custom-url-domain/main/index.md (hunk 3 above)
Comment (brentschaus-okta) [reply packages/@okta/vuepress-site/docs/guides/custom-url-domain/main/index.md:71]: Good call. I'll change it and in one other place, too.

File: packages/@okta/vuepress-site/docs/guides/custom-url-domain/main/index.md [hunk 5]
```diff
@@ -64,9 +64,27 @@ The third generation of the Okta Sign-In Widget doesn’t guarantee the stabilit
...
+
+  * Its start date shouldn't be in the future.
+
+  * Its expiration date can't be further than 10 years from now.
+
+  * It mustn't be self-signed.
+
+  * It must be a certificate for the actual domain.
+
+  * The public key must use the RSA algorithm.
```
Comment (timhowie-okta) [inline packages/@okta/vuepress-site/docs/guides/custom-url-domain/main/index.md:85]: is there a particular RSA algorithm that must be used?

**Review by timhowie-okta — APPROVED** [review]
lgtm

File: packages/@okta/vuepress-site/docs/guides/custom-url-domain/main/index.md [hunk 6]
```diff
@@ -64,9 +64,27 @@ The third generation of the Okta Sign-In Widget doesn’t guarantee the stabilit
...
+
+  * The public key isn't a certificate authority's (CA's).
```
[the remark this answers is not in the document]
Comment (brentschaus-okta) [reply packages/@okta/vuepress-site/docs/guides/custom-url-domain/main/index.md:87]: will clarify. thank you.

**Comment by acrolinxdevdocs-okta** [comment]
## Acrolinx Scorecards

Acrolinx checked the content changed here.

Please review the scorecards for each file and make the recommended
improvements.

| Article | Score | Issues | Scorecard | Status |
| ------- | ----- | ------ | --------- | ------ |
| [packages/@okta/vuepress-site/docs/guides/custom-url-domain/main/index.md](https://github.com/okta/okta-developer-docs/blob/87dd02050a85946e126643f538975c72d7712d2a/packages%2F%40okta%2Fvuepress-site%2Fdocs%2Fguides%2Fcustom-url-domain%2Fmain%2Findex.md) | 80 | 122 | [→](https://okta.acrolinx.cloud/api/v1/checking/scorecards/3caa9b92-8e9d-4e80-8d4a-10c732218e01) | [:white_check_mark:](# "Successfully processed") |

Successfully checked 1 of 1 documents.
See summary in [Content Analysis Dashboard](https://okta.acrolinx.cloud/api/batch/gen.github.cacbe478-b744-4e9b-a17b-83ed2eb7a920?csBaseUrl=https%3A%2F%2Fokta.acrolinx.cloud&X-Acrolinx-Client-Locale=en)

Reopen the pull request or push new changes to check again.

Depending on the Acrolinx server configuration, the
links expire after some time and you must have a login for the
Acrolinx server to access them again.

File: packages/@okta/vuepress-site/docs/guides/custom-url-domain/main/index.md (hunk 5 above)
Comment (gormanho-okta) [reply packages/@okta/vuepress-site/docs/guides/custom-url-domain/main/index.md:85]: The statement is correct - the certificate's public key must use RSA.

Here are all the validation messages we can return when validating the certificate: https://github.com/atko-eng/okta-core/blob/master/resources/src/main/resources/messages-untranslated.properties#L6331-L6354

Comment (brentschaus-okta) [reply packages/@okta/vuepress-site/docs/guides/custom-url-domain/main/index.md:85]: Thank you, Gorman :)

**Pull request merged**